integrated_channels/integrated_channel/exporters/learner_data.py

```diff
@@ -372,6 +532,58 @@ def _collect_certificate_data(self, enterprise_enrollment):
 
         return completed_date, grade, is_passing, percent_grade
 
+    def _collect_assessment_grades_data(self, enterprise_enrollment):
+        """
+        Collect a learner's assessment level grade data using an enterprise enrollment, from the Grades API.
+
+        Args:
+            enterprise_enrollment (EnterpriseCourseEnrollment): the enterprise enrollment record for which we need to
+            collect subsection grades data
+        Returns:
+            Dict:
+                {
+                    [subsection name]: {
+                        'grade_category': category,
+                        'grade': percent grade,
+                        'assessment_label': label,
+                        'grade_point_score': points earned on the assignment,
+                        'grade_points_possible': max possible points on the assignment,
+                        'subsection_id': subsection module ID
+                    }
+
+                    ...
+                }
+        """
+        if self.grades_api is None:
+            self.grades_api = GradesApiClient(self.user)
+
+        course_id = enterprise_enrollment.course_id
+        username = enterprise_enrollment.enterprise_customer_user.user.username
+        try:
+            assessment_grades_data = self.grades_api.get_course_assessment_grades(course_id, username)
+        except HttpNotFoundError as error:
+            if hasattr(error, 'response'):
+                response_content = error.response.json()  # pylint: disable=no-member
+                if response_content.get('error_code', '') == 'user_not_enrolled':
+                    return {}
+
+            return {}
+
```

Comment:
not clear why the error processing is here? it seems we always return {} regardless and there is no raise or anything

Reply:
oh, you know- this existed in the course learner data exporter. I don't think it's really necessary  ¯\_(ツ)_/¯

Reply:
Ahh, I see. The course learner data transmitter uses the error response content for logging reasons then returns the same thing. So yup, not needed. Unless we wanted to add more logging here.

Reply:
I'm mostly curious what we want to do in the case we don't hit that if condition (ie `if response_content.get('error_code', '') == 'user_not_enrolled':`). We seem to be saying that in this condition, we can just provide an empty dict, but otherwise.....something should happen. Either we just log and continue or we raise or something.

Reply:
I copied the behavior here from the similar `_collect_grades_data` method which does the following: 
```
        except HttpNotFoundError as error:
            # Grade not found, so we have nothing to report.
            if hasattr(error, 'response'):
                response_content = error.response.json()  # pylint: disable=no-member
                if response_content.get('error_code', '') == 'user_not_enrolled':
                    # This means the user has an enterprise enrollment record but is not enrolled in the course yet
                    LOGGER.info(
                        '[Integrated Channel] User is not enrolled in the course.'
                        ' Course: {course_id}, EnterpriseEnrollment: {enterprise_enrollment},'
                        ' Username: {username}'.format(
                            course_id=course_id,
                            username=username,
                            enterprise_enrollment=enterprise_enrollment.pk))
                    return None, None, None, None

            LOGGER.error('[Integrated Channel] Grades data not found.'
                         ' Course: {course_id}, EnterpriseEnrollment: {enterprise_enrollment},'
                         ' Username: {username}'.format(
                             course_id=course_id,
                             username=username,
                             enterprise_enrollment=enterprise_enrollment.pk))
            return None, None, None, None
```
The outcomes of both conditions yields the same return, but we log that there is an enterprise enrollment without a course enrollment. Since I was avoiding logging to make the PR more clear in its intent, I just removed the first condition. Having said that, I don't even think there's a reason for the first condition to return at all, except to not double log.  ¯\_(ツ)_/¯

Reply:
Yeah it's just in copying the behaviour you made an if condition that doesn't do anything. At least in that condition above it logs based on the condition. 

It's fine if we feel logs here would be useless, but then since it's not raising an exception, I think we should just get rid of this entire `if hasattr...` block.

Reply:
https://github.com/edx/edx-enterprise/pull/1048/files#diff-f488bfc03a35af19cc710b17a06ab3014e240abef4ad52114822c8b0e081b5caR562-R565
I already did, which is why I originally marked this as `resolved`. Unless you think logs here are necessary, in which case I'm happy to add it back in.